sdk/src/main/java/com/mapbox/maps/renderer/widget/BitmapWidgetRenderer.kt

```diff
@@ -0,0 +1,307 @@
+        Matrix.multiplyMM(mvpMatrix, 0, screenMatrix, 0, translateRotate, 0)
+
+        mvpMatrixBuffer = mvpMatrix.toFloatBuffer()
+
+        updateMatrix = false
+      }
+
+      GLES20.glUniformMatrix4fv(uniformMvpMatrix, 1, false, mvpMatrixBuffer)
+
+      textureFromBitmap()
+
+      GLES20.glActiveTexture(GLES20.GL_TEXTURE0)
+      GLES20.glBindTexture(GLES20.GL_TEXTURE_2D, textures[0])
+
+      GLES20.glUniform1i(uniformTexture, 0)
+
+      GLES20.glClearColor(1f, 1f, 1f, 1f)
+
+      GLES20.glEnableVertexAttribArray(attributeVertexPosition)
+        .also { GlUtils.checkError("glEnableVertexAttribArray") }
+
+      GLES20.glVertexAttribPointer(
+        attributeVertexPosition, COORDS_PER_VERTEX,
+        GLES20.GL_FLOAT, false,
+        VERTEX_STRIDE, vertexPositionBuffer
+      ).also { GlUtils.checkError("glVertexAttribPointer") }
+
+      GLES20.glEnableVertexAttribArray(attributeTexturePosition)
+        .also { GlUtils.checkError("glEnableVertexAttribArray") }
+
+      GLES20.glVertexAttribPointer(
+        attributeTexturePosition, COORDS_PER_VERTEX,
+        GLES20.GL_FLOAT, false,
+        VERTEX_STRIDE, texturePositionBuffer
+      ).also { GlUtils.checkError("glVertexAttribPointer") }
+
+      GLES20.glDrawArrays(GLES20.GL_TRIANGLE_STRIP, 0, VERTEX_COUNT)
+        .also { GlUtils.checkError("glDrawArrays") }
+
+      GLES20.glDisableVertexAttribArray(attributeVertexPosition)
+      GLES20.glDisableVertexAttribArray(attributeTexturePosition)
+      GLES20.glBindBuffer(GLES20.GL_ARRAY_BUFFER, 0)
+      GLES20.glUseProgram(0)
+    }
+    needRender = false
+  }
+
+  override fun deinitialize() {
+    if (program != 0) {
+      GLES20.glDisableVertexAttribArray(attributeVertexPosition)
+      GLES20.glDetachShader(program, vertexShader)
+      GLES20.glDetachShader(program, fragmentShader)
+      GLES20.glDeleteShader(vertexShader)
+      GLES20.glDeleteShader(fragmentShader)
+      GLES20.glDeleteTextures(textures.size, textures, 0)
+      GLES20.glDeleteProgram(program)
+      program = 0
+    }
+    needRender = false
+  }
+
+  private fun textureFromBitmap() {
+    if (!bitmap.isRecycled) {
+      GLES20.glGenTextures(1, textures, 0)
+      GLES20.glBindTexture(GLES20.GL_TEXTURE_2D, textures[0])
+      GLES20.glTexParameterf(
+        GLES20.GL_TEXTURE_2D,
+        GLES20.GL_TEXTURE_MIN_FILTER,
+        GLES20.GL_NEAREST.toFloat()
+      )
+      GLES20.glTexParameterf(
+        GLES20.GL_TEXTURE_2D,
+        GLES20.GL_TEXTURE_MAG_FILTER,
+        GLES20.GL_LINEAR.toFloat()
+      )
+      GLES20.glTexParameterf(
+        GLES20.GL_TEXTURE_2D,
+        GLES20.GL_TEXTURE_WRAP_S,
+        GLES20.GL_CLAMP_TO_EDGE.toFloat()
+      )
+      GLES20.glTexParameterf(
+        GLES20.GL_TEXTURE_2D,
+        GLES20.GL_TEXTURE_WRAP_T,
+        GLES20.GL_CLAMP_TO_EDGE.toFloat()
+      )
+      GLUtils.texImage2D(GLES20.GL_TEXTURE_2D, 0, bitmap, 0)
+      bitmap.recycle()
```

Comment:
Should we care about this here or the owner of bitmap should care about it?

Reply:
Refactored bitmap to be nullable instead.
Though caller anyway shouldn't recycle bitmap while we're not done with it. I'll drop a comment about that.